pkg/agent/flowexporter/connections/conntrack_connections.go

```diff
@@ -158,25 +158,31 @@ func (cs *ConntrackConnectionStore) addNetworkPolicyMetadata(conn *flowexporter.
 		egressOfID := binary.LittleEndian.Uint32(conn.Labels[4:8])
 		if ingressOfID != 0 {
 			policy := cs.networkPolicyQuerier.GetNetworkPolicyByRuleFlowID(ingressOfID)
+			rule := cs.networkPolicyQuerier.GetRuleByFlowID(ingressOfID)
 			if policy == nil {
 				// This should not happen because the rule flow ID to rule mapping is
 				// preserved for max(5s, flowPollInterval) even after the rule deletion.
 				klog.Warningf("Cannot find NetworkPolicy that has rule with ingressOfID %v", ingressOfID)
 			} else {
 				conn.IngressNetworkPolicyName = policy.Name
 				conn.IngressNetworkPolicyNamespace = policy.Namespace
+				conn.IngressNetworkPolicyType = flowexporter.PolicyTypeToUint8(policy.Type)
+				conn.IngressNetworkPolicyRuleName = rule.Name
```

Comment:
Sorry for the last minute comment, I just found we don't check if the rule is nil here, is it guarantted that rule is not nil when policy is not nil? Could a deletion happen between L160 and L161?

Reply:
I don't think that's guaranteed. Just added the nil check to be safe. Thanks!